### ISSUE TITLE
C++ version of MobX

### ISSUE THREAD

**Issue by robclouth** [mobxjs/mobx#554]
After using MobX for a while on web and app projects, I really, really want it for managing my state in C++ projects. I've seen a C++14 implementation of Redux...how doable would it be for MobX?

**Comment by mweststrate** [comment]
If you are using it in a single threaded environment it should be straight forward as there is no browser specific stuff. There is a lot of reflection / properties stuff in `types/`, but you could suffice by porting just the `core/` folder and leaving the observable types out initially expect for `ObservableValue` and create your own ones. (porting an older version might be simpler for starters, a lot of non-essential features where introduced at 2.2 and a lot of tiny core optimizations in 2.5). If you want to support threads, I think that should be doable as well, but you have to think it through :)

**Comment by mweststrate** [comment]
Closing this one for now. But let me know if somebody ports mobX to C# :)

**Comment by BlueSolei** [comment]
I think this library look close to MobX
[https://github.com/schlangster/cpp.react](url)

**Comment by jspuij** [comment]
@mweststrate Still interested in a C# version? I ported mobx to C# the last three weeks (called Cortex.Net). Complete with Blazor (webassembly) bindings. I've created the obligatory todomvc as a sample:

http://todomvc.spuij.nl/#/

It's MIT licenced. Before I open up the repository, I'd like to create the documentation. Would be nice to base it on the original MobX docs. Am I allowed to use the original docs as well?
